runtime/src/iree/hal/drivers/hip/native_executable.c

```diff
@@ -206,20 +212,25 @@ static iree_status_t iree_hal_hip_native_executable_flatbuffer_verify(
 }
 
 iree_status_t iree_hal_hip_native_executable_create(
-    const iree_hal_hip_dynamic_symbols_t* symbols, hipDevice_t device,
+    const iree_hal_hip_dynamic_symbols_t* symbols, uint32_t num_devices,
+    iree_hal_hip_per_device_information_t* devices,
     const iree_hal_executable_params_t* executable_params,
     iree_allocator_t host_allocator, iree_hal_executable_t** out_executable) {
   IREE_ASSERT_ARGUMENT(symbols);
   IREE_ASSERT_ARGUMENT(executable_params);
   IREE_ASSERT_ARGUMENT(out_executable);
+  if (num_devices < 1) {
+    return iree_make_status(IREE_STATUS_INVALID_ARGUMENT,
+                            "Need at least one device");
```

Comment:
Status messages start lower case (as other things are prepended). Try to write full sentences that explain the error.
```suggestion
                            "at least one device is required but none were provided");
```